### PR TITLE
Wait for event loop to no longer be alive on destruction

### PR DESCRIPTION
**Pull request by benh**

### CODE REVIEW

File: stout/event-loop.cc [hunk 1]
```diff
@@ -142,13 +142,26 @@ EventLoop::~EventLoop() {
...
-  }
+  CHECK(alive) << "should still have prepare and async handles to close";
+
+  do {
+    alive = uv_run(&loop_, UV_RUN_NOWAIT);
+
+    LOG_IF_EVERY_N(WARNING, alive, 100000)
+        << "destructing EventLoop with active handles:\n"
+        << [&]() {
+             // NOTE: lambda only gets executed every 100 instances
```
Comment (FolMing) [inline stout/event-loop.cc:158]: 100k?

**Review by FolMing — APPROVED** [review]
LGTM!

File: stout/event-loop.cc (hunk 1 above)
Comment (benh) [reply stout/event-loop.cc]: Thanks!

**Review by FolMing — APPROVED** [review]
Great stuff! Really like the way you try to simulate the output from `uv_print_all_handles`!

**Pull request merged**